sqlserver/datadog_checks/sqlserver/stored_procedures.py

```diff
@@ -0,0 +1,186 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import time
+
+from datadog_checks.base import is_affirmative
+from datadog_checks.base.utils.db.statement_metrics import StatementMetrics
+from datadog_checks.base.utils.db.utils import (
+    DBMAsyncJob,
+    default_json_event_encoding,
+)
+from datadog_checks.base.utils.serialization import json
+from datadog_checks.base.utils.tracking import tracked_method
+
+try:
+    import datadog_agent
+except ImportError:
+    from ..stubs import datadog_agent
+
+from datadog_checks.sqlserver.const import STATIC_INFO_ENGINE_EDITION, STATIC_INFO_VERSION
+
+DEFAULT_COLLECTION_INTERVAL = 60
+
+SQL_SERVER_PROCEDURE_METRICS_COLUMNS = [
+    "execution_count",
+    "total_worker_time",
+    "total_physical_reads",
+    "total_logical_writes",
+    "total_logical_reads",
+    "total_elapsed_time",
+    "total_spills",
+]
+
+# TODO: we are doing TOP here, we should use some ORDER BY
+PROCEDURE_METRICS_QUERY = """\
+SELECT TOP ({limit})
+    OBJECT_SCHEMA_NAME([object_id], [database_id]) as schema_name,
+    OBJECT_NAME([object_id], [database_id]) as procedure_name,
+    DB_NAME([database_id]) as database_name,
+    -- max(type_desc) as type_desc,
+    -- max(cached_time) as cached_time,
+    -- max(last_execution_time) as last_execution_time,
+    {procedure_metrics_columns}
+FROM sys.dm_exec_procedure_stats
+WHERE database_id <> 32767 -- reserved resource db
+GROUP BY object_id, database_id;
+"""
+
+
+def _row_key(row):
+    """
+    :param row: a normalized row from STATEMENT_METRICS_QUERY
+    :return: a tuple uniquely identifying this row
+    """
+    return "".join(
+        (
+            row.get('database_name'),
```

Comment:
I think we can do `row['database_name']` here since you have filtered out rows without `database_name` in later process https://github.com/DataDog/integrations-core/pull/15805/files#diff-129095abd7f622d5f90bce2c6b7441a6cfdcb8c53d5676e6c3d82624b9d39763R143